desc/compute/_stability.py

```diff
@@ -0,0 +1,276 @@
+from ._field import (
+    compute_magnetic_field_magnitude,
+    compute_boozer_magnetic_field,
+    compute_contravariant_current_density,
+)
+
+
+def compute_mercier_stability(
+    R_lmn,
+    Z_lmn,
+    L_lmn,
+    p_l,
+    i_l,
+    Psi,
+    R_transform,
+    Z_transform,
+    L_transform,
+    pressure,
+    iota,
+    data=None,
+):
+    """Compute the Mercier stability criterion.
+
+    Notes
+    -----
+        Implements equations 4.16 through 4.20 in M. Landreman & R. Jorge (2020)
+        doi:10.1017/S002237782000121X.
+
+    Parameters
+    ----------
+    R_lmn : ndarray
+        Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+    Z_lmn : ndarray
+        Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+    L_lmn : ndarray
+        Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+    p_l : ndarray
+        Spectral coefficients of p(rho) -- pressure profile.
+    i_l : ndarray
+        Spectral coefficients of iota(rho) -- rotational transform profile.
+    Psi : float
+        Total toroidal magnetic flux within the last closed flux surface (Wb).
+    R_transform : Transform
+        Transforms R_lmn coefficients to real space.
+    Z_transform : Transform
+        Transforms Z_lmn coefficients to real space.
+    L_transform : Transform
+        Transforms L_lmn coefficients to real space.
+    pressure : Profile
+        Transforms p_l coefficients to real space.
+    iota : Profile
+        Transforms i_l coefficients to real space.
+
+    Returns
+    -------
+    data : dict
+        Dictionary of ndarray, shape(num_nodes,) of Mercier criterion terms.
+        Keys are 'D_shear', 'D_current', 'D_well', 'D_geodesic', and 'D_Mercier'.
+
+    """
+    grid = R_transform.grid
+    data = compute_pressure(p_l, pressure, data=data)
+    data = compute_toroidal_flux_gradient(
+        R_lmn, Z_lmn, Psi, R_transform, Z_transform, data=data
+    )
+    data = compute_geometry(R_lmn, Z_lmn, R_transform, Z_transform, data=data)
+    data = compute_magnetic_field_magnitude(
+        R_lmn,
+        Z_lmn,
+        L_lmn,
+        i_l,
+        Psi,
+        R_transform,
+        Z_transform,
+        L_transform,
+        iota,
+        data=data,
+    )
+    data = compute_boozer_magnetic_field(
+        R_lmn,
+        Z_lmn,
+        L_lmn,
+        i_l,
+        Psi,
+        R_transform,
+        Z_transform,
+        L_transform,
+        iota,
+        data,
+    )
+    data = compute_contravariant_current_density(
+        R_lmn,
+        Z_lmn,
+        L_lmn,
+        i_l,
+        Psi,
+        R_transform,
+        Z_transform,
+        L_transform,
+        iota,
+        data,
+    )
+
+    # dS = |sqrt(g)| * |grad(rho)| * dt * dz
+    dS = jnp.abs(data["sqrt(g)"]) * data["|grad(rho)|"]
+
+    if check_derivs("D_shear", R_transform, Z_transform, L_transform):
+        data["D_shear"] = compress(
```

Comment:
I'd rather have the compress bit done on the objective side, so that all the compute functions return things that are broadcastable with each other

Reply:
I agree with Rory in that I think the internal compute_X functions should return things which are broadcastable to eachother, but I also agree we should have things like eq.compute("I") give unique values (unless specified to output on a specific grid maybe?)